Tribler/Test/Core/CreditMining/test_creditmining_sys.py

```diff
@@ -0,0 +1,376 @@
+
+        exist_torrent_url = "http://localhost:RANDOMPORT/existbuntu.torrent".replace(
+            "RANDOMPORT", str(self.file_server_port))
+        exist_infohash = sha1(exist_torrent_url).digest()
+
+        shutil.copyfile(TORRENT_FILE, os.path.join(self.boosting_manager.settings.credit_mining_path,
+                                      'rss-%s.torrent' % exist_infohash.encode('hex')))
+        self.setUpFileServer(self.file_server_port, files_path)
+
+        # now the rss should be at :
+        # http://localhost:port/test_rss_cm.xml
+        # which resides in sessiondir/http_torrent_files
+
+    def test_rss(self):
+        """
+        test rss source
+        """
+        url = 'http://localhost:%s/test_rss_cm.xml' %self.file_server_port
+        self.boosting_manager.add_source(url)
+
+        rss_obj = self.boosting_manager.get_source_object(url)
+        count = 0
+        while not rss_obj.ready and count < 10:
+            rss_obj = self.boosting_manager.get_source_object(url)
+            time.sleep(5)
+            count += 1
+        self.assertTrue(rss_obj.ready, "Not Ready")
+
+        count = 0
+        while len(rss_obj.torrents) < 1 and count < 10:
+            time.sleep(2)
+            count += 1
+
+        self.assertGreaterEqual(len(rss_obj.torrents), 1, "not enough torrent")
+
+        rss_obj.kill_tasks()
+
+        self.assertFalse(rss_obj.ready, "object still there")
+
+    def test_rss_unexist(self):
+        """
+        test unexist rss
+        """
+        url = 'http://localhost:%s/nothingness' %self.file_server_port
+        self.boosting_manager.add_source(url)
+
+        rss_obj = self.boosting_manager.get_source_object(url)
+        count = 0
+        while not rss_obj.feed_handle and count < 10:
+            rss_obj = self.boosting_manager.get_source_object(url)
+            time.sleep(3)
+            count += 1
+
+        self.assertFalse(rss_obj.ready, "Somehow can accept the url")
+        self.assertIsNotNone(rss_obj.feed_handle, "Environment not ready yet")
+
+        rss_obj = self.boosting_manager.get_source_object(url)
+
+        while rss_obj.feed_handle.get_feed_status()['updating']:
+            time.sleep(1)
+
+        import pprint
+        pprint.pprint(rss_obj.feed_handle.get_feed_status())
+        self.assertGreaterEqual(len(rss_obj.feed_handle.get_feed_status()['error']), 1, "not error")
+
+        rss_obj.kill_tasks()
+
+        self.assertFalse(rss_obj.ready, "object still there")
+
+    def tearDown(self):
+        super(TestBoostingManagerSysRSS, self).tearDown()
+
+
+class TestBoostingManagerSysDir(TestBoostingManagerSys):
+    """
+    testing class for directory source
+    """
+    def __init__(self, *argv, **kwargs):
+        super(TestBoostingManagerSysDir, self).__init__(*argv, **kwargs)
+
+    def setUp(self, autoload_discovery=True):
+        super(TestBoostingManagerSysDir, self).setUp()
+
+    def set_boosting_settings(self):
```

Comment:
If the commented-out code will not be added at a later point, this method can also be removed.